samples/subscriptions/createPushSubscription.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019-2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on
+ * subscriptions with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+'use strict';
+
+// Creates a new push subscription.
+function main(
```

Comment:
For each of these that take parameters - a preamble like this ensures the generated README and such are nice to use:

```js

// sample-metadata:
//   title: Export Assets
//   description: Export asserts to specified dump file path.
//   usage: node exportAssets.js <gs://my-bucket/my-assets.txt>
```

It would be nice to have one of these for each sample if possible.  For an example, see:
https://github.com/googleapis/nodejs-asset/blob/master/samples/exportAssets.js

Reply:
This is actually an interesting case for an unrelated reason: I'm pretty sure exceptions still won't be caught when running it from the command line. The main() function is async, but it's not calling await on the inner function, so it's essentially returning a Promise that immediately resolves. It might be something we should revisit in the style guide, because I'm not sure I see a good way to catch the errors from samples unless we tack a .catch() on the call to the inner function or await on it (in which case the sample snippet is weird if pasted elsewhere).

Back to the topic at hand: this is nice, I didn't know you could add these metadata comments. I'll do that.

Reply:
It's a really good point.  This is an underspecified part of the sample style guide that we should probably look at designing out further .. FYI @fhinkel

Reply:
If we don't await, don't we get an uncaught exception and abort?

Reply:
I made an issue to track this over there:
https://github.com/googleapis/nodejs-asset/issues/242

I'm going to resolve this conversation over here since the metadata bits got updated.